analyzebooks.py

```diff
@@ -0,0 +1,194 @@
+    AUTH: Connor Novak
+    MAIL: connor.novak@students.olin.edu
+    '''
+
+import string
+import operator
+import numpy as np
+import pdb
+
+def get_word_list(file_name):
+    '''
+        DESC: Reads specified project Gutenberg book, strips header comments,
+        punctuation, and whitespace, then returns lowercase list of words in book.
+        ARGS:
+        file_name - str - name of file where Gutenberk book text is stored
+        RTRN: list of lowercase words in book
+        '''
+    book = open(file_name, 'r')
+    lines = book.readlines()
+
+    curr_line = 0
+    # For every line that is before the start of the book, cut it out of lines
+    while lines[curr_line].find('START OF THIS PROJECT GUTENBERG EBOOK') == -1:
+        curr_line += 1
+    lines = lines[curr_line+1:]
+
+    curr_line = len(lines)
+    # For every line that is after the end of the book, cut it out of lines
+    while lines[curr_line-1].find('END OF THIS PROJECT GUTENBERG EBOOK') == -1:
+        curr_line -= 1
+    lines = lines[:curr_line-1]
+
+    # creates list of all words in all lines
+    words = []
+    for line in lines:
+        line = line.replace('\u2019','')
+        line = line.translate(str.maketrans('\u201c' + '\u201d' +string.punctuation,"                                  "))
+        words.extend(line.split())
+
+    # shifts words to all lowercase
+    for word in range(len(words)):
+        w = words[word]
+        w = w.lower()
+        words[word] = w
+
+    return words
+
+
+def get_top_n_words(word_list, n):
+    """
+        DESC: Returns n most frequently occurring words in a list of words
+        ARGS:
+        word_list - list - word list (assumed to all be in lower case with no
+        punctuation)
+        n - int - number of words to return (pass -1 to get all)
+        RTRN: list of n most frequently occurring words ordered from most
+        frequently to least frequently occurring
+        """
+    word_freq = {}
+    for word in word_list:
+        if (word in word_freq.keys()):
+            word_freq[word] += 1
+        else:
+            word_freq[word] = 1
+
+    sorted_word_freq = sorted(word_freq.items(), key=operator.itemgetter(1))
+    if(n == -1):
+        return sorted_word_freq
+    else:
+        return sorted_word_freq[-n-1:-1]
+
+
+def total_word_freq(booklist):
+    """
+        DESC: Generates word frequency list for all books in list
+        ARGS:
+        booklist - list - list of file names for books
+        RTRN: word frequency list for all books
+        """
+    wordlist = []
+    for book in booklist:
+        print('MSG: checking file: '+book)
+        wordlist.extend(get_word_list(book))
+    wordfreq = get_top_n_words(wordlist,-1)
+    return wordfreq
+
+
+def average_text(booklist):
+    """
+        DESC: Averages text frequencies of books in list
+        ARGS:
+        booklist - list - list of file names for books to average
+        RTRN: word frequency of the average book in the list
```

Comment:
It would be good to specify the data type of a return value just like what you did with the parameters.